File: CRPL.Data/Applications/ViewModels/OwnershipRestructure/OwnershipRestructureViewModel.cs
using CRPL.Data.StructuredOwnership;

namespace CRPL.Data.Applications.ViewModels;

public class OwnershipRestructureViewModel : ApplicationViewModel
{
    public List<OwnershipStake> CurrentStructure { get; set; }
    public List<OwnershipStake> ProposedStructure { get; set; }
    public BindStatus BindStatus { get; set; }
}

public class OwnershipRestructureViewModelWithoutAssociated : ApplicationViewModelWithoutAssociated
{
    
}

File: CRPL.Web/ClientApp/src/app/_Models/Applications/OwnershipRestructureViewModel.ts
import {OwnershipStake} from "../StructuredOwnership/OwnershipStake";
import {ApplicationViewModel} from "./ApplicationViewModel";

export interface OwnershipRestructureViewModel extends ApplicationViewModel
{
  CurrentStructure: OwnershipStake[];
  ProposedStructure: OwnershipStake[];
  BindStatus: BindStatus;
}

export enum BindStatus
{
  NoProposal,
  AwaitingVotes,
  Bound,
  Rejected
}
